Update documentation on bastion ssh configs to use the proxycommand values

eedgar opened the pull request.

eedgar committed:
Update documentation on bastion ssh configs to use the proxycommand values

punkrokk commented on docs/source/troubleshooting/ssh.rst:
```diff
@@ -85,7 +85,48 @@ runner boxes, and add the following configuration lines in ``/etc/st2/st2.conf``
+  chmod 600 /root/.ssh/config
+  chmod 600 /root/.ssh/id_rsa
+
+This is a sample ssh config that is known to work with bastion forwarding.
```
```suggestion
If you are looking to do ssh bastion forwarding, while allowing SSH to resolve automatically the correct keys based on hostname (eg. to dynamically support environments where a ssh hosts are set in your ssh_config file): 

This is a sample ssh config that is known to work with bastion forwarding.
```

eedgar commented on docs/source/troubleshooting/ssh.rst:
```diff
@@ -85,7 +85,48 @@ runner boxes, and add the following configuration lines in ``/etc/st2/st2.conf``
-  ssh_config_file_path = /home/stanley/.ssh/config
+  ssh_config_file_path = /root/.ssh/config
+
+Make sure your ssh config is in the same account as user running the st2action process.  If root is running 
```
I was testing against the docker-stackstorm checkout... if prod is running the daemon than changing the owner for all files to stanley makes sense.  You may want to still include that the docker all in run runs as root or keep it running as the staction command owner.

arm4b reviewed (changes requested):
Initially this looks like a SSH troubleshooting doc.

Wouldn't it be better to include anything related to custom bastion SSH config in a separated sub-topic or even another page?

@punkrokk Do you think we have a better place for such configuration example?

punkrokk commented:
@armab I'm not opposed to this. Let's see what happens with the PR (https://github.com/StackStorm/st2/pull/4881) and then we can update and make it happen.

LindsayHill committed:
Merge branch 'master' into bastion_ssh_config

blag commented on docs/source/troubleshooting/ssh.rst:
```diff
@@ -85,7 +85,48 @@ runner boxes, and add the following configuration lines in ``/etc/st2/st2.conf``
+
+This is a sample ssh config that is known to work with bastion forwarding.
+
+.. code-block:: ini
```
GitHub's [Linguist project](https://github.com/github/linguist) [supports syntax highlighting for SSH (client) configs](https://github.com/github/linguist/issues/3450), but you have to mark it as `ssh-config`:

```suggestion
.. code-block:: ssh-config
```

![SSH (client) config coloring screenshot](https://user-images.githubusercontent.com/597113/90189951-f682d100-dd72-11ea-87a3-225d9729d7c4.png)

And while you are at it, could you also fix this for line 160?

Leave line 84 alone though, since `/etc/st2/st2.conf` is just an INI file and is correctly marked as such. :+1:

cognifloyd replied:
I just pushed a fix for this.

cognifloyd replied:
Ah. We can't use `ssh-config` because this Sphinx uses pygments, not GitHub's linguist. So `ini` is the way to go.

cognifloyd committed:
Merge branch 'master' into bastion_ssh_config

cognifloyd committed:
Use correct code-block formatting

cognifloyd committed:
Clean up comment about bastion forwarding example

Co-authored-by: JP Bourget <jp.bourget@gmail.com>

cognifloyd committed:
leave stanley as the default user

cognifloyd committed:
reword ssh bastion forwarding description

cognifloyd commented:
I just reworded some of the prose about bastion forwarding. Hopefully it fits better in the troubleshooting section now. If not, I'm not sure where else it might go.
@armab and @punkrokk What do you think, is this good to merge now?

cognifloyd committed:
fix code-block formatting

cognifloyd committed:
Merge branch 'master' into bastion_ssh_config

cognifloyd committed:
Use ini instead of ssh-config for code blocks

We are using sphinx which uses pygments. pygments does not support
ssh-config format. github/linguist does, but that does not help us.

cognifloyd committed:
Merge branch 'master' into bastion_ssh_config

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StackStorm/st2docs?pullRequest=965) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/StackStorm/st2docs?pullRequest=965) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: eedgar<br/>:white_check_mark: cognifloyd<br/>:x: LindsayHill<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StackStorm/st2docs?pullRequest=965) it.</sub>

cognifloyd committed:
Merge branch 'master' into bastion_ssh_config
